layer/filestore.go

```diff
@@ -97,16 +97,20 @@ func (fm *fileMetadataTransaction) SetCacheID(cacheID string) error {
 	return ioutil.WriteFile(filepath.Join(fm.root, "cache-id"), []byte(cacheID), 0644)
 }
 
-func (fm *fileMetadataTransaction) TarSplitWriter() (io.WriteCloser, error) {
+func (fm *fileMetadataTransaction) TarSplitWriter(compressInput bool) (io.WriteCloser, error) {
 	f, err := os.OpenFile(filepath.Join(fm.root, "tar-split.json.gz"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		return nil, err
 	}
+	var wc io.WriteCloser
```

Comment:
Right, so the default would be to use only `fm.TarSplitWriter()` and for the migration use the helper fn.

Below we'd move the close wrapper to the helper, since that's only needed because of the compression, and return `f` directly (as an io.WriteCloser).